packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/helpers.ts

```diff
@@ -68,6 +68,23 @@ export function mergeMigrationMappingPropertyHashes(
   };
 }
 
+/**
+ * If `.kibana` and the version specific aliases both exists and
+ * are pointing to the same index. This version's migration has already
+ * been completed.
+ */
+export function versionMigrationCompleted(
+  currentAlias: string,
+  versionAlias: string,
+  aliases: Record<string, string>
```

Comment:
nit: for TS strictness, we could be explicit that some aliases could not exist:
```suggestion
  aliases: Record<string, string | undefined>
```